libraries/tock-register-interface/README.md

```diff
@@ -176,20 +176,90 @@ register_bitfields! [
         ERRORCOUNT OFFSET(6) NUMBITS(3) []
     ],
 
-    // In a simple case, offset can just be a number, and the number of bits
-    // is set to 1:
+    // Example using interrupt flags
+    /*
     InterruptFlags [
-        UNDES   10,
-        TXEMPTY  9,
-        NSSR     8,
-        OVRES    3,
-        MODF     2,
-        TDRE     1,
-        RDRF     0
+        UNDES   OFFSET(10) NUMBITS(1) [],
+        TXEMPTY OFFSET(9) NUMBITS(1) [],
+        NSSR    OFFSET(8) NUMBITS(1) [],
+        OVRES   OFFSET(3) NUMBITS(1) [],
+        MODF    OFFSET(2) NUMBITS(1) [],
+        TDRE    OFFSET(1) NUMBITS(1) [],
+        RDRF    OFFSET(0) NUMBITS(1) [], // Receive Data Register Full flag at bit 0
+
+        // Interrupt mode that can be LevelTriggered, EdgeTriggered, or PulseTriggered
+        INTERRUPT_MODE OFFSET(4) NUMBITS(2) [
+            LevelTriggered = 0,
+            EdgeTriggered = 1,
+            PulseTriggered = 2,
+        ]
```

Comment:
Sorry, my comment could have been phrased better.

I think it's important for us to showcase the short-hand notation in `InterruptFlags`, so to leave that example mostly unmodified. However, to avoid us needing to show it's expansion as well (as it's a different syntax), we should instead show that this short-hand notation is equivalent to the more verbose one above. Here's how I imagine this might look like:

```rust
    // In a simple case, offset can just be a number, and the number of bits
    // is set to 1:
    InterruptFlags [
        // This is equivalent to writing
        // UNDES OFFSET(10) NUMBITS(1) [],
        UNDES   10,
        TXEMPTY  9,
        NSSR     8,
        OVRES    3,
        MODF     2,
        TDRE     1,
        RDRF     0
    ]
```